### ISSUE TITLE
Creating Checkboxes for Rows Seclection

### ISSUE THREAD

**Issue by bucketpress** [fooplugins/FooTable#503]
@steveush , I've managed to create the checkboxes functionality with the following:

Firstly by declaring these in the columns with other column data:

```
"columns" : [
    {...},
    {"name":"_selection", "title" : "Selection", 'visible': false, "sortable": false, "filterable":false},
    {"name":"_checkbox", "title" : "<input type='checkbox' class='global-checkbox'>", 'visible': true, "sortable": false, "filterable":false}
]
```

'_selection' values will be used for the actual filtering. Checkboxes will toggle the row value of '_selection' as 'True' or 'False'. The original intention was to set the data-value attr of the td that holds the checkbox. But I couldn't find an efficient method of setting the data-value during runtime. Also taking into the fact that I'm loading my data via ajax with a json response. I've set '_selection' as `"filterable":false` because I do not want it to appear in the default search dropdown.

The row values for the checkbox in each row is `'<input type="checkbox" class="row-checkbox">'`

So if there is a good way to set the data-value of the checkbox td on the fly, there won't be a need for an additional '_selection' column.

Next, I needed to capture the events and callbacks for when the global checkbox, and a row's checkbox is toggled:

```
//On Global Checkbox Toggle, set all rows' checkbox to checked and toggle _selection values
$table.on("change", ".global-checkbox", function() {
    var newValues = {};
    if ( $(this).prop("checked") ) {
        newValues._selection = 'True';
        newValues._checkbox = '<input type="checkbox" class="row-checkbox" checked>';
    } else {
        newValues._selection = 'False';
        newValues._checkbox = '<input type="checkbox" class="row-checkbox">';
    }
    $.each(ft.rows.all, function( index, values ) {
        ft.rows.update(index, newValues, false);
    });
    ft.draw();
});

//On Single Row Checkbox Toggle
$table.on('click', '.row-checkbox', function() {
    var newValues = {};
    var row = $(this).closest('tr').data('__FooTableRow__');
    if ( $(this).prop('checked') ) {
        //Prepare Values
        newValues._selection = 'True';
        newValues._checkbox = '<input type="checkbox" class="row-checkbox" checked>';
    } else {
        //Prepare Values
        newValues._selection = 'False';
        newValues._checkbox = '<input type="checkbox" class="row-checkbox">';

        //Toggle globalCheckbox if checked
        if ( $globalCheckbox.prop('checked') ) {
             $globalCheckbox.prop('checked', false);
        }
    }
    row.val(newValues, true);
});
```

Note that the above 'row.val(...)' and ft.rows.update(...) can work properly only with the change as mentioned in #499 . Else we should prepare the newValues that includes all key:value pairs that we want to keep for each row.

Then I created a custom Filter Dropdown to filter '_selection' based on the values of 'True' and 'False' as described in the docs: https://fooplugins.github.io/FooTable/docs/examples/advanced/filter-dropdown.html

For my use case, as I have to load data via ajax multiple times. Whenever there's an ajax call, I need to run:
`$table.find('.global-checkbox').eq(0).prop('disabled', true)`
to make sure that the user doesn't toggle the global checkbox while new data has not loaded, because I need to ensure that the new set of rows' '_selection' and '_checkbox' value and 'checked' property follows the 'checked' property of the global checkbox.

That means on an ajax success, I iterate through the response data like this before loading the data into the table:

```
data.forEach(function(item, index, array) {
    if ( $globalCheckbox.prop('checked') ) {
        data[index]._selection = 'True';
        data[index]._checkbox = '<input type="checkbox" class="row-checkbox" checked >';
    } else {
        data[index]._selection = 'False';
        data[index]._checkbox = '<input type="checkbox" class="row-checkbox">';
    }
});
```

Then I make sure that while initialising footable I have this option so that after the table is ready, the global checkbox is accessible again:

```
"on": {
    "postdraw.ft.table": function(e, ft) {
            $globalCheckbox.prop('disabled', false);
    }
}
```

**Comment by bucketpress** [comment]
@steveush , one question though - is there any way or option for the custom filter dropdown to be applied to some tables and not on others. The example shown applies to all footables.

**Comment by steveush** [comment]
@bucketpress take a look at the [Custom Filter UI](http://fooplugins.github.io/FooTable/docs/examples/advanced/custom-filter-ui.html) example. The custom dropdown you've already looked at shows you how to modify the filter component but as you stated that modification is a global change for all tables on the page. If you want to modify the filters for only a single table then you will need to take a look at the custom filter UI example and just not hide the original but rather append to it.

**Comment by steveush** [comment]
Hi @bucketpress,

With regards to your question about how to have a custom filter dropdown applied to only a single table. In version 3.1.0 I have added in a [components option](http://fooplugins.github.io/FooTable/docs/getting-started.html#components) which allows you to load a custom component on a per table basis. I've updated the [custom dropdown filter example](http://fooplugins.github.io/FooTable/docs/examples/advanced/filter-dropdown.html), specifically step 6, which shows when to make use of this option.

Thanks
